enterprise/server/raft/store/store.go

```diff
@@ -1062,6 +1058,97 @@ func (s *Store) TransferLeadership(ctx context.Context, req *rfpb.TransferLeader
 	return &rfpb.TransferLeadershipResponse{}, nil
 }
 
+func (s *Store) waitForReplicaOpen(ctx context.Context, clusterID uint64) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			if rd := s.lookupRange(clusterID); rd != nil {
+				return nil
+			}
+			time.Sleep(10 * time.Millisecond)
+		}
+	}
+}
+
+func (s *Store) SnapshotCluster(ctx context.Context, clusterID uint64) error {
```

Comment:
I'd add a comment to clarify that this a local operation (i.e. it doesn't affect other nodes)

Reply:
Done.